src/routes/v2/authenticated/review.ts

```diff
@@ -685,7 +706,43 @@ export class ReviewsRouter {
       possibleReviewRequest
     )
 
-    return res.status(200).json()
+    if (isIsomerError(possibleReviewRequest)) {
+      return res.status(404).json({ message: possibleReviewRequest.message })
+    }
+
+    // Step 4: Check if the user is a reviewer of the RR
+    const { reviewers } = possibleReviewRequest
+    const isReviewer = _.some(
+      reviewers,
+      (user) => user.email === userWithSiteSessionData.email
+    )
+
+    if (!isReviewer) {
+      return res.status(401).json({
+        message: "Only reviewers can approve Review Requests!",
+      })
+    }
+
+    // Step 5: Approve review request
+    // NOTE: We are not checking for existence of PR
+    // as the underlying Github API returns 404 if
+    // the requested review could not be found.
+    await this.reviewRequestService.approveReviewRequest(possibleReviewRequest)
+
+    // Step 6: Create notifications
+    const collaborators = await this.collaboratorsService.list(siteName)
+    collaborators.forEach(async (user: User & { SiteMember: SiteMember }) => {
```

Comment:
see above comment wrt `forEach` and async loop

Reply:
b9449d4